### PR TITLE
Filter out connection failures

### PR DESCRIPTION
**Pull request by asonnino**
Filter out connection failures. This may happen when the the transaction is not sequenced by consensus and is instead dropped.

### CODE REVIEW

**Comment by bmwill** [comment]
In practice wouldn't we want a queue of txns and to continue retrying to submit them instead of dropping them?

File: sui/tests/shared_objects_tests.rs [hunk 1]
```diff
@@ -66,8 +66,14 @@ async fn submit_shared_object_transaction(
...
+            let reply = match &result {
+                Ok(_) => Some(result),
+                Err(SuiError::ConsensusConnectionBroken(..)) => None,
+                Err(..) => Some(result),
+            };
+            replies.push(reply)
         }
+
         if replies.iter().any(|x| x.is_some()) {
             // Remove all `ConsensusConnectionBroken` replies.
```
Comment (bmwill) [inline sui/tests/shared_objects_tests.rs:78]: This comment seems to indicate the filtering out of those errors happens below?

Comment (asonnino) [reply sui/tests/shared_objects_tests.rs:78]: no it doesn't, but your question indicates that I should write this code better :p let me have another try

**Comment by asonnino** [comment]
> In practice wouldn't we want a queue of txns and to continue retrying to submit them instead of dropping them?

That is why we have the loop line 57

File: sui/tests/shared_objects_tests.rs (hunk 1 above)
Comment (asonnino) [reply sui/tests/shared_objects_tests.rs:70]: fixed

**Review by bmwill — APPROVED** [review]

**Pull request merged**